airbyte-workers/src/main/java/io/airbyte/workers/storage/StateStore.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.storage;
+
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.State;
+import io.airbyte.config.storage.CloudStorageConfigs.GcsConfig;
+import io.airbyte.config.storage.CloudStorageConfigs.MinioConfig;
+import io.airbyte.config.storage.CloudStorageConfigs.S3Config;
+import io.airbyte.config.storage.DefaultGcsClientFactory;
+import io.airbyte.config.storage.DefaultS3ClientFactory;
+import io.airbyte.config.storage.MinioS3ClientFactory;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Leverages the CloudDocumentStore. The root directory is state and then each activityRunId is the
```

Comment:
make sure I understand, this state refers to a state output by a temporal activity?

Reply:
yes!